search/src/KBMOSearch.cpp

```diff
@@ -45,14 +60,80 @@ void KBMOSearch::enableCorr(std::vector<float> pyBaryCorrCoeff)
     } 
 }
     
-void KBMOSearch::enableGPUFilter(std::vector<float> pyPercentiles, 
+void KBMOSearch::enableGPUFilter(std::vector<float> pyPercentiles,
                                  float pySigmaGCoeff,
                                  float pyMinLH)
 {
-    gpuFilter = true;
-    percentiles = pyPercentiles;
-    sigmaGCoeff = pySigmaGCoeff;
-    minLH = pyMinLH;
+    params.doFilter = true;
+    params.sGL_L = pyPercentiles[0];
+    params.sGL_H = pyPercentiles[1];
+    params.sigmaGCoeff = pySigmaGCoeff;
+    params.minLH = pyMinLH;
+}
+
+void KBMOSearch::enableGPUEncoding(int psiNumBytes, int phiNumBytes)
+{
+    assert(psiNumBytes == 1 || psiNumBytes == 2);
+    assert(phiNumBytes == 1 || phiNumBytes == 2);
+
+    params.encodeImg = true;
+    params.psiNumBytes = psiNumBytes;
+    params.phiNumBytes = phiNumBytes;
+
+    // Set the bounds for psi and phi based on the bounds
+    // from the data.
+    preparePsiPhi();
+    float minPsi = FLT_MAX;
+    float maxPsi = -FLT_MAX;
+    float minPhi = FLT_MAX;
+    float maxPhi = -FLT_MAX;
+
+    const int num_images = psiImages.size();
+    assert(num_images == phiImages.size());
+
+    for (int t = 0; t < num_images; ++t)
+    {
+        const std::vector<float>& psiRef = psiImages[t].getPixels();
```

Comment:
I think if we wanted to reduce duplication of code, we could just call fillPsiAndPhiVects here instead of redoing the loop (might make it easier to cache the results as well in the future if we find that loading the pixel vects is getting expensive)

Reply:
I think we can even take this one step further by breaking the function into two: the "enable" function and a later "set bounds" function that just takes psiVect and phiVect directly.